g2p_programs/tests/test_cycle_manager_new_cycle.py

```diff
@@ -0,0 +1,145 @@
+        first_cycle = program_manager.new_cycle()
+
+        self.assertEqual(first_cycle.start_date, datetime.now().date())
+        self.assertEqual(first_cycle.end_date, datetime(2023, 1, 3).date())
+        self.assertEqual(first_cycle.sequence, 1)
+
+        second_cycle = program_manager.new_cycle()
+
+        self.assertEqual(second_cycle.start_date, datetime(2023, 1, 4).date())
+        self.assertEqual(second_cycle.end_date, datetime(2023, 1, 23).date())
+        self.assertEqual(second_cycle.sequence, 2)
+
+    def test_new_cycle_yearly(self):
+        # create and test a yearly cycle with 1 cycle duration
+
+        program_manager_data = {
+            "name": "Yearly",
+            "rrule_type": "yearly",
+            "cycle_duration": 1,
+        }
+
+        program_manager = self.create_program_manager(**program_manager_data)
+
+        first_cycle = program_manager.new_cycle()
+
+        self.assertEqual(first_cycle.start_date, datetime.now().date())
+        self.assertEqual(first_cycle.end_date, datetime(2023, 12, 14).date())
+        self.assertEqual(first_cycle.sequence, 1)
+
+        second_cycle = program_manager.new_cycle()
+
+        self.assertEqual(second_cycle.start_date, datetime(2023, 12, 15).date())
+        self.assertEqual(second_cycle.end_date, datetime(2024, 12, 14).date())
+        self.assertEqual(second_cycle.sequence, 2)
+
+    def test_new_cycle_monthly_month_by_date(self):
+        # create and test a monthly cycle with 1 cycle duration every 10th day of the month
+
+        program_manager_data = {
+            "name": "Monthly every 10th",
+            "rrule_type": "monthly",
+            "cycle_duration": 1,
+            "month_by": "date",
+            "day": 10,
+        }
+
+        program_manager = self.create_program_manager(**program_manager_data)
+
+        first_cycle = program_manager.new_cycle()
+
+        self.assertEqual(first_cycle.start_date, datetime.now().date())
+        self.assertEqual(first_cycle.end_date, datetime(2023, 1, 9).date())
+        self.assertEqual(first_cycle.sequence, 1)
+
+        second_cycle = program_manager.new_cycle()
+
+        self.assertEqual(second_cycle.start_date, datetime(2023, 1, 10).date())
+        self.assertEqual(second_cycle.end_date, datetime(2023, 2, 9).date())
+        self.assertEqual(second_cycle.sequence, 2)
+
+    def test_new_cycle_monthly_month_by_day(self):
+        # create and test a monthly cycle with 1 cycle duration every first monday of the month
+
+        program_manager_data = {
+            "name": "Monthly every first monday of the month",
+            "rrule_type": "monthly",
+            "cycle_duration": 1,
+            "month_by": "day",
+            "byday": "1",
+            "weekday": "MON",
+        }
+
+        program_manager = self.create_program_manager(**program_manager_data)
+
+        first_cycle = program_manager.new_cycle()
+
+        self.assertEqual(first_cycle.start_date, datetime.now().date())
```

Comment:
This should not be the case as `datetime.now()` is not the first Monday of the month, no?
Also do not use `datetime.now()` but hardcode the dates or use constants